agents/node/vcxagent-core/test/utils/faber.js

```diff
@@ -137,6 +151,31 @@ module.exports.createFaber = async function createFaber () {
     return agencyMessages
   }
 
+  async function _downloadConnectionRequests () {
+    logger.info('Faber is going to download connection requests')
+    let agencyMessages = await vcxAgent.serviceConnections.getAllMessages()
```

Comment:
Good question....maybe we could make it even a bit different
```
const request = await publicAgent.downloadConnectionRequest()
const connection = await createWithConnectionRequest(request)
connection.updateState()
```
have
```
const connection = await publicAgent.tryCreateConnectionFromConnectionRequest()
connection.updateState()
```
so the public agent would try to find connection request, and if found, it would update message in agency and return the connection. This would be basically equivalent to how `.updateState()` operates.

Otherwise
```
const request = await publicAgent.downloadConnectionRequest()
```
would have to return request message, but in "agency format", eg. including `uid`, so you could manually update the message state. But I'd rather go for the first option for now.

Reply:
About further filters - one thing which comes to my mind is how to delete non-ConnectionRequest messages. It *should not* happen that we have other messages on public agent, but if someone for whatever reason sends there something else, we should have way to clean up these things. It could be either convenient Rust API like `public_agent_clean_up` which update status of all non-ConnectionRequests as processed. 
Or it could be a generic API, like `public_agent_download_messages` which would return messages in the agency format (eg. including msgUid, status etc.) and it would be up to caller to call this with a proper filter and update messages in agency. I tend to lean toward the more convenient "cleanUp" API here again.

Reply:
As discussed, rejecting non-request messages when downloading now.